### ISSUE TITLE
Support for multiple numeric types

### ISSUE THREAD

**Issue by boxtown** [boxtown/statrs#17]
Currently statrs only supports f64 but I'd like to at the minimum extend that to f32 and possibly other numeric types as well (especially for things in the `statistics` module). The [num](https://crates.io/crates/num) crate might be worth looking into but I'm hesistant about introducing the dependency when it might make it's way in to the standard library at some point.

**Comment by vks** [comment]
I don't think `num` will move to the standard library. In fact, it has been moved out of it.

**Comment by boxtown** [comment]
@vks I was under the impression that a float trait would make it's way into the stdlib once [associated constants](https://users.rust-lang.org/t/replacement-trait-for-std-num-float/827/4) is implemented and stable? I'm not saying I refuse to use `num` just hesistant for now

**Comment by vks** [comment]
Some traits might be moved. For now you could just depend on `num_traits`, which is a fairly lightweight dependency.

**Comment by boxtown** [comment]
Yeah it just depends on when I get to this. Trying to port statistics::Statistics to Iterator is being a HUGE pain in my side

**Comment by boxtown** [comment]
This has been a really huge effort and pain with in my opinion not enough tangible benefits. Hate to do this but punting on this until the future if ever, and will restart v0.4.0 with other more useful targets

**Comment by bluss** [comment]
num-trait's interfaces are far from ideal. Basically needs a redesign, or entirely new interfaces to deal with float ←→ float conversions.

**Comment by YeungOnion** [comment]
`num_traits` has fallible conversion between floats with `NumCast` trait, think this is good to close now. Thoughts?
